python/tests/server/test_worker.py

```diff
@@ -335,21 +341,22 @@ async def test_cancel_idempotency():
         await _process(w.setup())
 
         p1_done = None
+        input1 = PredictionInput({"sleep": 0.5})
 
-        async for event in w.predict({"sleep": 0.5}, poll=0.01):
+        async for event in w.predict(input1, poll=0.01):
             # We call cancel a WHOLE BUNCH to make sure that we don't propagate
             # any of those cancelations to subsequent predictions, regardless
             # of the internal implementation of exceptions raised inside signal
             # handlers.
             for _ in range(100):
-                w.cancel()
+                w.cancel(input1.id)
```

Comment:
Yikes, does this mean we're relying on `Worker.predict` to modify its arguments in order to communicate prediction ID back to the caller? That seems very hacky.

Reply:
no? the prediction ID must be fixed before it makes it to Worker.predict. it is autogenerated if not set on PredictionRequest and then you're supposed to PredictionInput.from_request(request...)